src/Chapter3.hs

```diff
@@ -971,6 +1099,28 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Weekday = Monday 
+             | Tuesday 
+             | Wednesday 
+             | Thursday 
+             | Friday 
+             | Saturday 
+             | Sunday  deriving (Show, Read, Eq, Ord, Enum)
+
+isWeekend :: Weekday -> Bool
+isWeekend Saturday = True
+isWeekend Sunday   = True
+isWeekend _        = False
+
+nextDay :: Weekday -> Weekday
+nextDay Sunday = Monday
+nextDay day = succ day
+
+daysToParty :: Weekday -> Int
+daysToParty day =  (fromEnum Friday - fromEnum day + numDays) `mod` numDays
+    where numDays = length (enumFrom Monday) 
```

Comment:
If you want to have a list of all entities, there is actually a [common pattern](http://hackage.haskell.org/package/relude-0.7.0.0/docs/Relude-Extra-Enum.html#v:universe) for this in Haskell
```suggestion
    where numDays = length [mibBound .. maxBound] 
```
Though, since the number of days in a week rarely changes, it's enough to `mod` directly by `7` :slightly_smiling_face: 

Also, adding `numDays` is not necessary, since `mod` already behaves properly for negative numbers:

```haskell
ghci> mod (-1) 7
6
```

Reply:
Ohh, I tried it via
```Haskell
Prelude> -1 `mod` 7
-1
```
and forgot about operator precedences :man_facepalming:

Reply:
Ah, yes, Haskell precedence rules indeed can be confusing. A simple rule: when in doubt, put into `()`